src/main/java/io/stargate/sgv2/jsonapi/service/resolver/CreateCollectionCommandResolver.java

```diff
@@ -192,13 +197,35 @@ private CreateCollectionCommand.Options.VectorSearchConfig validateVectorOptions
 
     Integer vectorDimension = vector.dimension();
     VectorizeConfig service = vector.vectorizeConfig();
+    String sourceModel = vector.sourceModel();
+    String metric = vector.metric();
+
+    // decide sourceModel and metric value
+    if (sourceModel != null) {
+      if (metric == null) {
+        // (1) sourceModel is provided but metric is not - set metric to cosine or dot_product based
+        // on the map
+        metric = SUPPORTED_SOURCES.get(sourceModel).getMetric();
+      }
+      // (2) both sourceModel and metric are provided - do nothing
+    } else {
+      if (metric != null) {
+        // (3) sourceModel is not provided but metric is - set sourceModel to 'other'
+        sourceModel = SourceModel.OTHER.getSourceModel();
+      } else {
+        // (4) both sourceModel and metric are not provided - set sourceModel to 'other' and metric
+        // to 'cosine'
+        sourceModel = SourceModel.OTHER.getSourceModel();
+        metric = SimilarityFunction.COSINE.getMetric();
+      }
+    }
 
     if (service != null) {
       // Validate service configuration and auto populate vector dimension.
       vectorDimension = validateVectorize.validateService(service, vectorDimension);
       vector =
           new CreateCollectionCommand.Options.VectorSearchConfig(
-              vectorDimension, vector.metric(), vector.vectorizeConfig());
+              vectorDimension, metric, sourceModel, vector.vectorizeConfig());
     } else {
```

Comment:
Do we want to optimize the `sourceModel` for service? For example, if the user only specifies `openai` and `text-embedding-3-small` in the `service`, do we want to optimize the `sourceModel` to `openai-v3-small`. Currently, we will use `other` as the default value.